src/main/kotlin/no/nav/familie/ks/sak/kjerne/tilbakekreving/HentFagsystemsbehandlingRequestConsumer.kt

```diff
@@ -0,0 +1,55 @@
+package no.nav.familie.ks.sak.kjerne.tilbakekreving
+
+import no.nav.familie.kontrakter.felles.objectMapper
+import no.nav.familie.kontrakter.felles.tilbakekreving.HentFagsystemsbehandlingRequest
+import no.nav.familie.kontrakter.felles.tilbakekreving.HentFagsystemsbehandlingRespons
+import no.nav.familie.kontrakter.felles.tilbakekreving.Ytelsestype
+import no.nav.familie.ks.sak.config.KafkaConfig
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import org.slf4j.LoggerFactory
+import org.springframework.context.annotation.Profile
+import org.springframework.kafka.annotation.KafkaListener
+import org.springframework.kafka.support.Acknowledgment
+import org.springframework.stereotype.Service
+import java.util.concurrent.CountDownLatch
+
+@Service
+@Profile("!integrasjonstest & !dev-postgres-preprod")
+class HentFagsystemsbehandlingRequestConsumer(private val fagsystemsbehandlingService: FagsystemsbehandlingService) {
+
+    private val logger = LoggerFactory.getLogger(this::class.java)
+    private val secureLogger = LoggerFactory.getLogger("secureLogger")
+
+    var latch: CountDownLatch = CountDownLatch(1)
```

Comment:
Er ikke så kjent med CountDownLatch, men vil dette funke?
Slik jeg tolker det kan CountDownLatch kun telle ned én gang og man må bruke `.await()` for å blokke.

Kunne man ha brukt en mutex i stedet?
https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.sync/-mutex/
typ:

```
val mutex = Mutex(1)

fun listen(){
  if (mutex.tryLock()){
     try {
       ...logikk...
    } finally {
      mutex.unlock() 
    }
  } else { return }
}

Reply:
Kan egentlig flere tråder kjøre denne funkjsonen? Dette vil vel ikke begrense på tvers av podder? 🤔

Reply:
@halvorbmundal har ikke så mye kunnskap på dette heller. Kopierte denne deler fra ba-sak og jeg tror både EF og BA bruker dette.

Reply:
@halvorbmundal fjernet countDown implementasjon foreløpig fordi jeg ikke husker hvorfor vi lagde det i utgangspunkt. Kan legges til senere ved behov

Reply:
Klarer ikke å se at vi se at EF bruker KafkaListener: https://github.com/navikt/familie-ef-sak/search?q=KafkaListener

Virker ikke som vi har noen trådsikring i baks-mottak i alle fall: https://github.com/navikt/familie-baks-mottak/blob/4ba6083b9ade0b8d17ca0430d0c0c8ebddbcf7c7/src/main/kotlin/no/nav/familie/baks/mottak/hendelser/Journalf%C3%B8ringHendelseAivenConsumer.kt

Kan se ut som vi ikke trenger CountDownLatch